src/roles/meta.js

```diff
@@ -0,0 +1,27 @@
+
+class MetaRole {
+
+  recharge(creep) {
```

Comment:
Not sure why we need a base class for one function that seems pretty standalone. Also, I don't see the reason to have a base class at all, given the way roles work in this system. These are just helper functions, could be free functions in a module and callable from anywhere someone has a Creep.

Reply:
I definitely plan on expanding this out into other things though, particularly with regards to helper functions with spawning and boosting.

Reply:
Right, but the thing I'm wondering about here is, if the class doesn't have any state, why do those methods need to go onto a base class, and not just a helper lib?

Sort of a stylistic nitpick, though.